src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -2738,3 +2741,139 @@ def test_named_and_wildcard_use_var(f2008_parser):
+            in str(excinfo.value))
+
+    # find_or_create_symbol method with invalid scope location
+    with pytest.raises(ValueError) as excinfo:
+        _ = _find_or_create_imported_symbol(alpha, alpha.name, scope_limit=alpha)
+    assert ("The scope_limit node 'Reference[name:'alpha']' provided to "
+            "_find_or_create_imported_symbol() is not an ancestor of this "
+            "node 'Reference[name:'alpha']'." in str(excinfo.value))
+
+    # With a visibility parameter
+    sym = _find_or_create_imported_symbol(alpha, "very_private",
+                                      visibility=Symbol.Visibility.PRIVATE)
+    assert sym.name == "very_private"
+    assert sym.visibility == Symbol.Visibility.PRIVATE
+    assert sym is container.symbol_table.lookup("very_private",
+                                                scope_limit=container)
+
+def test_find_or_create_imported_symbol_2():
+    ''' Check that the _find_or_create_imported_symbol() method creates new
+    symbols when appropriate. '''
+    # Create some suitable PSyIR from scratch
+    symbol_table = SymbolTable()
+    symbol_table.add(DataSymbol("tmp", REAL_TYPE))
+    kernel1 = KernelSchedule.create("mod_1", SymbolTable(), [])
+    container = Container.create("container_name", symbol_table,
+                                 [kernel1])
+    xvar = DataSymbol("x", REAL_TYPE)
+    xref = Reference(xvar)
+    assign = Assignment.create(xref, Literal("1.0", REAL_TYPE))
+    kernel1.addchild(assign)
+    assign.parent = kernel1
+    # We have no wildcard imports so there can be no symbol named 'undefined'
+    with pytest.raises(SymbolError) as err:
+        _ = _find_or_create_imported_symbol(assign, "undefined")
+    assert "No Symbol found for name 'undefined'" in str(err.value)
+    # We should be able to find the 'tmp' symbol in the parent Container
+    sym = _find_or_create_imported_symbol(assign, "tmp")
+    assert sym.datatype.intrinsic == ScalarType.Intrinsic.REAL
+    # Add a wildcard import to the SymbolTable of the KernelSchedule
+    new_container = ContainerSymbol("some_mod")
+    new_container.wildcard_import = True
+    kernel1.symbol_table.add(new_container)
+    # Symbol not in any container but we do have wildcard imports so we
+    # get a new symbol back
+    new_symbol = _find_or_create_imported_symbol(assign, "undefined")
+    assert new_symbol.name == "undefined"
+    assert isinstance(new_symbol.interface, UnresolvedInterface)
+    assert type(new_symbol) == Symbol
+    assert "undefined" not in container.symbol_table
+    assert kernel1.symbol_table.lookup("undefined") is new_symbol
+
+
+def test_nemo_find_container_symbol(parser):
+    ''' Check that find_or_create_symbol() works for the NEMO API when the
+    searched-for symbol is declared in the parent module. '''
+    from fparser.common.readfortran import FortranFileReader
+    from psyclone.psyir.nodes import BinaryOperation
```

Comment:
pylint complains that BinaryOperation has already been imported.

Reply:
There are pycode style errors about indentation for various lines and lines too long (2808 and 2815).

Reply:
Fixed all